store/auth.go

```diff
@@ -158,3 +160,40 @@ func ReadStoreToken() (*StoreToken, error) {
 
 	return &readStoreToken, nil
 }
+
+// RequestPackageAccessMacaroon requests a macaroon for accessing package data from the ubuntu store
+func RequestPackageAccessMacaroon() (string, error) {
+	emptyJsonData := "{}"
+	req, err := http.NewRequest("POST", myappsPackageAccessAPI, strings.NewReader(emptyJsonData))
+	req.Header.Set("accept", "application/json")
+	req.Header.Set("content-type", "application/json")
+	if err != nil {
+		return "", err
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	// check return code, error on anything !200
+	if !httpStatusCodeSuccess(resp.StatusCode) {
+		// unexpected result, bail
+		return "", fmt.Errorf("failed to get store macaroon: %v (%v)", resp.StatusCode, resp)
+	}
+
+	dec := json.NewDecoder(resp.Body)
+	responseData := map[string]string{}
+	if err := dec.Decode(&responseData); err != nil {
+		return "", err
+	}
+
+	macaroon, available := responseData["macaroon"]
+	if !available {
+		return "", fmt.Errorf("missing store macaroon: %v (%v)", resp.StatusCode, resp)
```

Comment:
`return "", fmt.Errorf("store returned empty package access macaroon")`

Similarly, showing something like `200 ({ 0  0 0 map[] <nil> 0 [] false map[] <nil> <nil>})` won't be very useful.